src/GraphQL/Types/GraphTypesLookup.cs

```diff
@@ -363,7 +362,6 @@ private void HandleField(IComplexGraphType parentType, FieldType field, TypeColl
             if (applyNameConverter)
             {
                 field.Name = NameConverter.NameForField(field.Name, parentType);
-                NameValidator.ValidateName(field.Name);
```

Comment:
This removes validation of names after they have been converted by the NameConverter.  This is the whole point of #1840 -- "Validate names after they are mapped with INameConverter".  So basically this PR reverts #1840

Reply:
Why? Validation happens inside `field.Name` setter. This is the point - to drill external validation from many places and keep it in one place.

Reply:
So that schema has been validated after the specified name converter has been applied.  So that a valid schema is enforced, while allowing a custom name converter.

> to drill external validation from many places and keep it in one place.

No, it is in **only one place now**.  You're moving it from one place (`GraphTypesLookup.HandleField`) and moving it to another place (`Field.Name` / `Argument.Name`).  And this new place does not properly validate the schema if someone uses a custom name converter.  You are creating a bug; the very same bug that #1840 was written to address.

I will grant you that the current place is not ideal for debugging field naming issues.  However what's the worst thing that can happen?  Someone will try to start their program, and it will still tell them what field is incorrectly named.  We can make the error a bit more specific to say which graph type contains the invalid field/argument name.  It simply won't highlight the invalid line of code in the debugger.  But it won't let an invalid schema be generated.

Now if the one check is insufficient for the needs of the program, then we can add another.  But if you can only pick one spot to put the check, then it's in the proper location now.  Otherwise, we are right back to where we started before #1840 was merged -- it's being checked in the wrong spot.  Which is why #1840 was titled "Validate names **after** they are mapped with INameConverter".  #1840 did not say anything about consolidating the number of checks, when in fact, that's what it did do.  It **was** scattered in multiple places in the code, like the field builders and stuff, and **now** it is in only one place: `GraphTypesLookup.HandleField`.

Reply:
> No, it is in only one place now. 

OK. Let's say so, but now I still can do that: `myField.Name = "$-#!"`

> You're moving it from one place (GraphTypesLookup.HandleField) and moving it to another place (Field.Name / Argument.Name).

OK. From one to another. But that new place handles all set cases. It doesn't matter when initializing the schema, before or after it.

> And this new place does not properly validate the schema if someone uses a custom name converter. 

What do you mean '_does not properly validate_' ? An exception from setter occurring at the earliest stages of schema building when doing something like `myField.Name = "$-#!"`? In this case, one can use `NameValidator.Validation` delegate. In general, it is better to fix the reason why such `$-#!` names are set so that the error does not occur.

That is, there are two situations when names are converted - when a 'good' name is converted to some other 'good' name (applying conventions) and when a 'bad' name is converted to a 'good' one (I have not encountered this in practice). I assume that you are only talking about the bug #1840 for the second case. For the first case, everything works as before. Right?

Reply:
@Shane32 Are you still sure about your statement https://github.com/graphql-dotnet/graphql-dotnet/pull/1964/files#r514275928 ? Validation occurs in setters.